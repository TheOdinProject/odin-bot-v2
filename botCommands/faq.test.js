const command = require('./faq');
const { generateMentions } = require('./mockData');

describe('?faq', () => {
  it.each([['?faq'], [' ?faq'], ['?faq @odin-bot'], [' ?faq @odin-bot']])(
    "'%s' triggers the callback",
    (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    },
  );

  it.each([
    ['?fEhq'],
    ['fye'],
    ['?f'],
    ['?fq'],
    ['?f a q'],
    ['```function("?faq", () => {}```'],
    ['?FAQ'],
    [''],
    [' '],
    ['? '],
    ['@odin-bot ? faq'],
    ['?fafaq'],
  ])("'%s' does not trigger the callback", (string) => {
    expect(command.regex.test(string)).toBeFalsy();
  });

  it.each([
    ['thats a ?faq'],
    ['that gets asked a lot ?faq give this a read'],
    ['hey @odin-bot, ?faq'],
    ['?@odin-bot ^ ?me ?faq ?test$*'],
  ])("%s' - the command can be anywhere in the string", (string) => {
    expect(command.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['@user?faq'],
    ['watchaknowabout?faq'],
    ['?faqsgrealkdmsfalnd'],
    ['?faq?'],
    ['?faq*'],
    ['?faq...'],
  ])(
    "'%s' - command should be its own word/group - no leading or trailing characters",
    (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    },
  );
});

describe('callback', () => {
  it('should return the correct output', async () => {
    expect(await command.cb(generateMentions(0))).toMatchSnapshot();
    expect(await command.cb(generateMentions(1))).toMatchSnapshot();
    expect(await command.cb(generateMentions(2))).toMatchSnapshot();
    expect(await command.cb(generateMentions(3))).toMatchSnapshot();
  });
});
