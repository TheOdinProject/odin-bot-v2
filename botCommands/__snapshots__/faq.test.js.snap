// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`callback should return the correct output 1`] = `
Object {
  "embeds": Array [
    Object {
      "color": 13407555,
      "description": "**The definition of insanity is answering the same question over and over again when we have an [#faq](https://discord.com/channels/505093832157691914/823266307293839401/823266549912829992)!  Help us stay sane by giving it a read.**",
      "title": "Frequently Asked Questions",
    },
  ],
}
`;

exports[`callback should return the correct output 2`] = `
Object {
  "embeds": Array [
    Object {
      "color": 13407555,
      "description": "**The definition of insanity is answering the same question over and over again when we have an [#faq](https://discord.com/channels/505093832157691914/823266307293839401/823266549912829992)! <@1>, help us stay sane by giving it a read.**",
      "title": "Frequently Asked Questions",
    },
  ],
}
`;

exports[`callback should return the correct output 3`] = `
Object {
  "embeds": Array [
    Object {
      "color": 13407555,
      "description": "**The definition of insanity is answering the same question over and over again when we have an [#faq](https://discord.com/channels/505093832157691914/823266307293839401/823266549912829992)! <@1> and <@2>, help us stay sane by giving it a read.**",
      "title": "Frequently Asked Questions",
    },
  ],
}
`;

exports[`callback should return the correct output 4`] = `
Object {
  "embeds": Array [
    Object {
      "color": 13407555,
      "description": "**The definition of insanity is answering the same question over and over again when we have an [#faq](https://discord.com/channels/505093832157691914/823266307293839401/823266549912829992)! <@1>, <@2>, and <@3>, help us stay sane by giving it a read.**",
      "title": "Frequently Asked Questions",
    },
  ],
}
`;
